--- a/src/x.c	2026-08-30 01:07:38.436887165 -0400
+++ b/src/x.c	2026-08-30 01:07:44.258179078 -0400
@@ -511,6 +511,7 @@
     }
 
     if (con->fullscreen_mode
+        || con_border_style(con) == BS_NONE
         || (!con_is_floating(con) && config.smart_gaps == SMART_GAPS_ON && con_num_visible_children(con->parent) <= 1)) {
         xcb_shape_mask(conn, XCB_SHAPE_SO_SET, XCB_SHAPE_SK_BOUNDING, con->frame.id, 0, 0, XCB_NONE);
         xcb_shape_mask(conn, XCB_SHAPE_SO_SET, XCB_SHAPE_SK_CLIP, con->frame.id, 0, 0, XCB_NONE);
